src/helpers.c

```diff
@@ -874,3 +877,167 @@ struct list *files_from_bundles(struct list *bundles)
 
 	return files;
 }
+
+void concat_str_array(char **output, char *const argv[])
+{
+	int size = 0;
+
+	for (int i = 0; argv[i]; i++) {
+		size += strlen(argv[i]) + 1;
+	}
+
+	*output = malloc(size + 1);
+	if (!*output) {
+		fprintf(stderr, "Failed to allocate %i bytes\n", size);
+		assert(0);
+	}
+	strcpy(*output, "");
+	for (int i = 0; argv[i]; i++) {
+		strcat(*output, argv[i]);
+		strcat(*output, " ");
+	}
+}
+
+int system_argv(char *const argv[])
+{
+	int child_exit_status;
+	pid_t pid;
+	int status = -1;
+
+	pid = fork();
+
+	if (pid == 0) { /* child */
+		execvp(*argv, argv);
+		/* Next line must not be reached */
+		assert(0);
+	} else if (pid < 0) {
+		fprintf(stderr, "Failed to fork a child process\n");
+		assert(0);
+	} else {
+		pid_t ws = waitpid(pid, &child_exit_status, 0);
+
+		if (ws == -1) {
+			fprintf(stderr, "Failed to wait for child process\n");
+			assert(0);
+		}
+
+		if (WIFEXITED(child_exit_status)) {
+			status = WEXITSTATUS(child_exit_status);
+		} else {
+			fprintf(stderr, "Child process didn't exit\n");
+			assert(0);
```

Comment:
Uhh, this is just for testing, right? Whatever happens to subprocesses, we should really be sure that we can stop updating and not throw an error and abort.